src/shortest_path/mod.rs

```diff
@@ -77,6 +77,17 @@ pub fn graph_dijkstra_shortest_paths(
     weight_fn: Option<PyObject>,
     default_weight: f64,
 ) -> PyResult<PathMapping> {
+    if !graph
+        .node_indices()
+        .nodes
+        .iter()
+        .any(|&node| node == source)
```

Comment:
I think we can simplify this to:
```suggestion
    if !graph.graph.contains_node(start)
```

Reply:
Note that you will need to move the if statement after the variable is defined